tests/LinkTest.php

```diff
@@ -6,10 +6,76 @@ class LinkTest extends CopilotTagTest
 {
     public function expectedWrites()
     {
+        $assetMarkdown = "\n\n[#image: /photos/123ID]|||some caption|||\n";
```

Comment:
I prefer the idea of generating this value rather than having it inline to ensure we are using an up-to-date value. What was the rationale of not doing so here?

Reply:
**TL;DR: I avoid executing code from one module to test another, just in case they're both equally wrong and your tests still pass.**

I just wanted to be more explicit about what the value is of asset markdown. When I was looking through the tests in Tugboat for this, I just wanted to know what this value actually was. It's not a big deal, but I feel it's more explicit.

How often is the markdown spec going to change?

You're right in that if it changes we'll also have to update it here, and that it could end up being stale. However, I can foresee some situation where we're doing `$assetMarkdown = new Asset('a', 'b', 'c');` and (for some reason) `$assetMarkdown` gets set to null or an empty string or something (bad import, old code, edge cases, typos, blah) and all of our tests are passing but they're not testing what we think they are.

Again this isn't super likely either but there felt like code smell. I generally try and avoid executing code from one module to test another, just in case they're both equally wrong and your tests still pass.

Reply:
@graycodes I wasn't concerned about the spec changing, more concerned about the whitespace being added in `Embed::write` changing. We are adding 2 leading newlines and 1 trailing at the moment, but that might not always be the case, and we just have to remember to update this string literal if we change the behaviour of `Embed`.

I see what you mean though. I don't mind either way really.

Reply:
@graycodes can I suggest a compromise - we generate the input but write the expected output as a string literal?

Reply:
I guess I'll update once we create the asset tag

Reply:
@graycodes I don’t see the point of `Asset` and it’s not in the spec. `Embed` will do.

Reply:
ah okay